### PR TITLE
finished the 4 functions of task 2, please review

### PR DESCRIPTION
**Pull request by xuanwgit**

### CODE REVIEW

File: extract_data.py [hunk 1]
```diff
@@ -54,8 +63,116 @@ def get_stock_financials(stock:str) -> pd.DataFrame:
...
         "Stock"]]
 
+
     # Return the collected data
     return transposed_financials
 
 
+### Task 2
+# 1. add a function called get_exchange_rate to extract_data.py so it can download fx rate for us
+def get_exchange_rate(from_currency:str, to_currency:str, interval:str):
```
Comment (hgao62) [inline extract_data.py]: add return type

File: extract_data.py [hunk 2]
```diff
@@ -54,8 +63,116 @@ def get_stock_financials(stock:str) -> pd.DataFrame:
...
+                      close prices.
+    """
+    fx_rate_ticker = f"{from_currency}{to_currency}=X"
+    fx_rates = yf.download(fx_rate_ticker, period="ytd", interval=interval)
+    fx_rates = fx_rates.reset_index()
+    fx_rates["From Currency"] = from_currency
+    fx_rates["To Currency"] = to_currency
+    fx_rates["Ticker"] = f"{from_currency}{to_currency}=X"
+    fx_rates = fx_rates[["Date","Ticker","From Currency", "To Currency", "Open", "High", "Low", "Close", "Adj Close"]]
+    print(fx_rates.head())  # Inspect data
```
Comment (hgao62) [inline extract_data.py]: remove print

File: extract_data.py [hunk 3]
```diff
@@ -54,8 +63,116 @@ def get_stock_financials(stock:str) -> pd.DataFrame:
...
+    return fx_rates
+    
+print(get_exchange_rate("USD", "GBP", "1d"))
```
Comment (hgao62) [inline extract_data.py]: remove print

File: extract_data.py [hunk 4]
```diff
@@ -54,8 +63,116 @@ def get_stock_financials(stock:str) -> pd.DataFrame:
...
+        stock (str): The stock ticker symbol (e.g., 'MSFT' for Microsoft).
+
+    Returns:
+        None: The function prints the trading currency directly, but does not return a value.
+    """
+    stock_info = yf.Ticker(stock)
+    stock_currency = stock_info.fast_info.get("currency")
+    print(f"The trading currency of {stock_info} is: {stock_currency}")
+
+print(get_stock_currency_code("MSFT"))
```
Comment (hgao62) [inline extract_data.py]: same here

File: extract_data.py [hunk 5]
```diff
@@ -54,8 +63,116 @@ def get_stock_financials(stock:str) -> pd.DataFrame:
...
+        pd.DataFrame: A DataFrame containing news articles, with columns for stock symbol, UUID, title, publisher, 
+                      publishing time, type of article, and a link to the full article.
+    """
+    stock_info = yf.Ticker(stock)
+    stock_news = stock_info.get_news()
+    stock_df = pd.DataFrame(stock_news)
+    stock_df["Stock"] =stock
+    stock_df.columns = stock_df.columns.str.capitalize()
+    stock_df = stock_df[["Stock","Uuid","Title", "Publisher", "Link", "Providerpublishtime", "Type"]]
+    print(stock_df)
```
Comment (hgao62) [inline extract_data.py]: same print statement

File: extract_data.py [hunk 6]
```diff
@@ -54,8 +63,116 @@ def get_stock_financials(stock:str) -> pd.DataFrame:
...
+        None: The function prints the trading currency directly, but does not return a value.
+    """
+    stock_info = yf.Ticker(stock)
+    stock_currency = stock_info.fast_info.get("currency")
+    print(f"The trading currency of {stock_info} is: {stock_currency}")
+
+print(get_stock_currency_code("MSFT"))
+
+#3. add function called get_news to extract_data.py so we can get relevant news belongs to that company
+def get_news(stock:str):
```
Comment (hgao62) [inline extract_data.py]: add return type

File: extract_data.py [hunk 7]
```diff
@@ -54,8 +63,116 @@ def get_stock_financials(stock:str) -> pd.DataFrame:
...
+    fx_rates = fx_rates[["Date","Ticker","From Currency", "To Currency", "Open", "High", "Low", "Close", "Adj Close"]]
+    print(fx_rates.head())  # Inspect data
+    return fx_rates
+    
+print(get_exchange_rate("USD", "GBP", "1d"))
+print(get_exchange_rate("USD", "GBP", "1d").info())
+
+
+# 2. add a function called get_stock_currency_code so that we know what currency this stock belongs to
+def get_stock_currency_code(stock:str):
```
Comment (hgao62) [inline extract_data.py]: add return type

File: extract_data.py [hunk 8]
```diff
@@ -54,8 +63,116 @@ def get_stock_financials(stock:str) -> pd.DataFrame:
...
+        pd.DataFrame: The DataFrame containing the normalized stock data, with rounded numerical values 
+                      and the 'Date' column renamed to 'Trade_Date'.
+    """
+    stock_history_df = get_stock_history(stock)
+    columns_to_round = ["Open", "High", "Low", "Close"]
+    stock_history_df[columns_to_round] = stock_history_df[columns_to_round].round(2)
+    # Rename the "Date" column to "Trade_Date"
+    stock_history_df.rename(columns={"Date": "Trade_Date"}, inplace=True)
+    print(stock_history_df)
+print(normalize_stock_data("MSFT"))
```
Comment (hgao62) [inline extract_data.py]: remove print

**Comment by plantepro** [comment]
Updated based on the comment, please check.

**Pull request merged**